src/content_working_group/lib.rs

```diff
@@ -0,0 +1,2422 @@
+            };
+
+            CuratorOpeningById::<T>::insert(new_curator_opening_id, new_opening_by_id);
+
+            // Update NextCuratorOpeningId
+            NextCuratorOpeningId::<T>::mutate(|id| *id += <CuratorOpeningId<T> as One>::one());
+
+            // Trigger event
+            Self::deposit_event(RawEvent::CuratorOpeningAdded(new_curator_opening_id));
+        }
+
+        /// Begin accepting curator applications to an opening that is active.
+        pub fn accept_curator_applications(origin, curator_opening_id: CuratorOpeningId<T>)  {
+
+            // Ensure lead is set and is origin signer
+            Self::ensure_origin_is_set_lead(origin)?;
+
+            // Ensure opening exists in this working group
+            // NB: Even though call to hiring modul will have implicit check for
+            // existence of opening as well, this check is to make sure that the opening is for
+            // this working group, not something else.
+            let (curator_opening, _opening) = Self::ensure_curator_opening_exists(&curator_opening_id)?;
+
+            // Attempt to begin accepting applicationsa
+            // NB: Combined ensure check and mutation in hiring module
+            ensure_on_wrapped_error!(
+                hiring::Module::<T>::begin_accepting_applications(curator_opening.opening_id)
+                )?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Trigger event
+            Self::deposit_event(RawEvent::AcceptedCuratorApplications(curator_opening_id));
+        }
+
+        /// Begin reviewing, and therefore not accepting new applications.
+        pub fn begin_curator_applicant_review(origin, curator_opening_id: CuratorOpeningId<T>) {
+
+            // Ensure lead is set and is origin signer
+            let (_lead_id, _lead) = Self::ensure_origin_is_set_lead(origin)?;
+
+            // Ensure opening exists
+            // NB: Even though call to hiring modul will have implicit check for
+            // existence of opening as well, this check is to make sure that the opening is for
+            // this working group, not something else.
+            let (curator_opening, _opening) = Self::ensure_curator_opening_exists(&curator_opening_id)?;
+
+            // Attempt to begin review of applications
+            // NB: Combined ensure check and mutation in hiring module
+            ensure_on_wrapped_error!(
+                hiring::Module::<T>::begin_review(curator_opening.opening_id)
+                )?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Trigger event
+            Self::deposit_event(RawEvent::BeganCuratorApplicationReview(curator_opening_id));
+        }
+
+        /// Fill opening for curator
+        pub fn fill_curator_opening(
+            origin,
+            curator_opening_id: CuratorOpeningId<T>,
+            successful_curator_application_ids: BTreeSet<CuratorApplicationId<T>>
```

Comment:
Does an empty set need to be handled in a special way?
Would have to look at behavior of hiring module when calling `fill_opening` with empty application set.